engine/runtime/src/main/java/org/enso/interpreter/runtime/Context.java

```diff
@@ -463,4 +468,31 @@ public TruffleLogger getLogger(Class<?> klass) {
   public long clockTick() {
     return clock.getAndIncrement();
   }
+
+  /** Return the {@code Standard.Base.Data.Time.Date} constructor.
+   *
+   * @return optional with {@link AtomConstructor} for the date, if it can be found
+   */
+  public Optional<AtomConstructor> getDateConstructor() {
+    if (date == null) {
+      CompilerDirectives.transferToInterpreterAndInvalidate();
+      ensureModuleIsLoaded("Standard.Base.Data.Time.Date");
+      Optional<Module> dateModule = findModule("Standard.Base.Data.Time.Date");
+      if (dateModule.isPresent()) {
+        try {
+          InteropLibrary iop = InteropLibrary.getUncached();
```

Comment:
I'd rather use the strongly-typed API of `Module` to get at this atom constructor, rather than the stringly-typed interop library with typechecks. `module.getScope().getLocalConstructor("Date");` replaces the whole try-catch here.

Reply:
Done in [1252ec0](https://github.com/enso-org/enso/pull/3374/commits/1252ec0c8da7bfb2af8a21cee124510824663398), but not by using `getLocalConstructor`. That method returns `associatedType` and that's not the right constructor. Possibly related to the _mess of symbols_ as recently discussed also with @ekmett .